src/main/java/decodes/cwms/resevapcalc/SolarFlux.java

```diff
@@ -0,0 +1,408 @@
+     *  1. Thin Ci/Cs (denoted as cloud type 1)
+     *  2. Thick Ci/Cs (denoted as cloud type 2)
+     *  3. As/Ac (denoted as cloud type 3)
+     *  4. Low cloud (denoted as cloud type 4 or 5)   
+     */
+    private static final double[][] R2 =
+    {
+        { .25674,.42111,.61394,.69143 },
+        { -.18077,-.04002,-.01469,-.14419 },
+        { -.21961,-.51833,-.17400,-.05100 },
+        { .25272,.40540,.14215,.06682 }
+    };
+    
+    /*  tclr=b0 + b1*cosz + b2*cosz^2 + b3*cosz^3, where in the
+     *  following array the rows are b0, b1, b2 and b3, and
+     *  the columns correspond to
+     *   1. Highest layer
+     *   2. Middle layer
+     *   3. Lowest layer
+     *   4. Smoke/fog (Not used in this model)
+     */
+    private static final double[][] T1 =
+    {
+        { .76977,.69318,.68679,.55336 },
+        { .49407,.68227,.71012,.61511 },
+        { -.44647,-.64289,-.71463,-.29816 },
+        { .11558,.17910,.22339,-.06663 }
+    };
+    
+    /* tcld=b0 + b1*cosz + b2*cosz^2 + b3*cosz^3, where in the
+     * following array the rows are b0, b1, b2 and b3, and
+     * the columns correspond to
+     *  1. Thin Ci/Cs (denoted as cloud type 1)
+     *  2. Thick Ci/Cs (denoted as cloud type 2)
+     *  3. As/Ac (denoted as cloud type 3)
+     *  4. Low cloud (denoted as cloud type 4 or 5)   
+     */
+    private static final double[][] T2 =
+    {
+        { .63547,.43562,.23865,.15785 },
+        { .35229,.26094,.20143,.32410 },
+        { .08709,.36428,-.01183,-.14458 },
+        { -.22902,-.38556,-.07892,.01457 }
+    };
+
+    /* W=c0 + c1*cosz + c2*fk + c3*fk*cosz + c4*cosz^2 + c5*fk^2,
+     * where in the following array the rows are co,c1,c2,c3,c4
+     * and c5, and the columns correspond to
+     *  1. Thin Ci/Cs (denoted as cloud type 1)
+     *  2. Thick Ci/Cs (denoted as cloud type 2)
+     *  3. As/Ac (denoted as cloud type 3)
+     *  4. Low cloud (denoted as cloud type 4 or 5)   
+     */    
+    private static final double[][] WTX =
+    {
+        { 0.675,1.552,1.429,1.512 },
+        { -3.432,-1.957,-1.207,-1.176 },
+        { 1.929,-1.762,-2.008,-2.160 },
+        { 0.842,2.067,0.853,1.420 },
+        { 2.693,0.448,0.324,-0.032 },
+        { -1.354,0.932,1.582,1.422 }
+    };
+
+
+    //double gmt_offset;
+    double _diffuse;
+    double _direct;
+    double _sdown;
+
+    double ZEN;
+    double SOLAR;
+    
+    public void solflx(Date currentTime, double gmtOffset,
+                       double longitude,
+                       double latitude, CloudCover[] cloudCover )
+    {
+        EvapUtilities.DoubleContainer solzen = new EvapUtilities.DoubleContainer();
+        EvapUtilities.DoubleContainer sdowndc = new EvapUtilities.DoubleContainer();
+        EvapUtilities.DoubleContainer cosz = new EvapUtilities.DoubleContainer();
+        double frad,dhr;
+        double sdown, direct, diffuse;
+        int doy,hr;
+
+        Calendar calendar = Calendar.getInstance();
```

Comment:
So unlikely, but since you're setting the timezone here at least I've remember that not everyone uses the GregorianCalendar. I know Thailand uses a different calendar. While I don't believe we have any users there, or at least known that may use this algorithms immediately, not a good assumption to make the the current "instance". will be the one we expect for these.